lib/src/cwrapper.cc

```diff
@@ -0,0 +1,40 @@
+#include <facter/cwrapper.hpp>
+
+#include <facter/facts/collection.hpp>
+#include <facter/util/config.hpp>
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <sstream>
+
+uint8_t get_default_facts(char **result) {
+    try {
+        // NB: ssh resolver cannot be blocked
+        facter::facts::collection facts {{},      // blocklist set
+                                         {},      // ttls map (resolvers - ttl)
+                                         true};   // ignore_cache flag
+
+        // The boolean arg is meant to avoid including ruby facts
+        facts.add_default_facts(false);
```

Comment:
Why are we skipping external, environment, and ruby facts in this wrapper? If this is meant to be a generic wrapper, I'd expect it to either follow Facter's default behavior (which is to collect all of those), or to provide option flags for which things to collect.